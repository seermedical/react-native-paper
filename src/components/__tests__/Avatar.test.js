import * as React from 'react';
import { StyleSheet } from 'react-native';
import renderer from 'react-test-renderer';
import { fireEvent, render } from 'react-native-testing-library';
import * as Avatar from '../Avatar/Avatar.tsx';
import { red500 } from '../../styles/colors';

const styles = StyleSheet.create({
  bgColor: {
    backgroundColor: red500,
  },
});

it('renders avatar with text', () => {
  const tree = renderer.create(<Avatar.Text label="XD" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with text and custom size', () => {
  const tree = renderer.create(<Avatar.Text size={96} label="XD" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with text and custom background color', () => {
  const tree = renderer
    .create(<Avatar.Text style={styles.bgColor} label="XD" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with text and custom colors', () => {
  const tree = renderer
    .create(<Avatar.Text color="#FFFFFF" label="XD" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with icon', () => {
  const tree = renderer.create(<Avatar.Icon icon="information" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with icon and custom background color', () => {
  const tree = renderer
    .create(<Avatar.Icon style={styles.bgColor} icon="information" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders avatar with image', () => {
  const tree = renderer
    .create(<Avatar.Image source={{ src: 'avatar.png' }} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

describe('AvatarImage listener', () => {
  const onListenerMock = jest.fn();
  const { getByTestId } = render(
    <Avatar.Image
      testID={'avatar-image'}
      onError={onListenerMock}
      onLayout={onListenerMock}
      onLoad={onListenerMock}
      onLoadEnd={onListenerMock}
      onLoadStart={onListenerMock}
      onProgress={onListenerMock}
    />
  );

  it('onError should be called', () => {
    fireEvent(getByTestId('avatar-image'), 'onError');
    expect(onListenerMock).toHaveBeenCalled();
  });

  it('onLayout should be called', () => {
    fireEvent(getByTestId('avatar-image'), 'onLayout');
    expect(onListenerMock).toHaveBeenCalled();
  });

  it('onLoad should be called', () => {
    fireEvent(getByTestId('avatar-image'), 'onLoad');
    expect(onListenerMock).toHaveBeenCalled();
  });

  it('onLoadEnd should be called', () => {
    fireEvent(getByTestId('avatar-image'), 'onLoadEnd');
    expect(onListenerMock).toHaveBeenCalled();
  });

  it('onLoadStart should be called', () => {
    fireEvent(getByTestId('avatar-image'), 'onLoadStart');
    expect(onListenerMock).toHaveBeenCalled();
  });

  it('onProgress should be called', () => {
    fireEvent(getByTestId('avatar-image'), 'onProgress');
    expect(onListenerMock).toHaveBeenCalled();
  });
});
