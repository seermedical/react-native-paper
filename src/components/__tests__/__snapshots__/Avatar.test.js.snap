// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders avatar with icon 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    accessibilityElementsHidden={true}
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    selectable={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "color": "#ffffff",
          "fontSize": 38.4,
        },
      ]
    }
  >
    □
  </Text>
</View>
`;

exports[`renders avatar with icon and custom background color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f44336",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    accessibilityElementsHidden={true}
    importantForAccessibility="no-hide-descendants"
    pointerEvents="none"
    selectable={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "color": "#ffffff",
          "fontSize": 38.4,
        },
      ]
    }
  >
    □
  </Text>
</View>
`;

exports[`renders avatar with image 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "src": "avatar.png",
      }
    }
    style={
      Object {
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      }
    }
  />
</View>
`;

exports[`renders avatar with text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 32,
            "lineHeight": 32,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;

exports[`renders avatar with text and custom background color 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#f44336",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 32,
            "lineHeight": 32,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;

exports[`renders avatar with text and custom colors 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 32,
        "height": 64,
        "width": 64,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#FFFFFF",
            "fontSize": 32,
            "lineHeight": 32,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;

exports[`renders avatar with text and custom size 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#6200ee",
        "borderRadius": 48,
        "height": 96,
        "width": 96,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "color": "#ffffff",
            "fontSize": 48,
            "lineHeight": 48,
          },
          undefined,
        ],
      ]
    }
  >
    XD
  </Text>
</View>
`;
